wayland-protocols/wlr-foreign-toplevel-management-unstable-v1.xml

```diff
@@ -255,5 +255,25 @@
         actually changes, this will be indicated by the state event.
       </description>
     </request>
+
+    <!-- Version 3 additions -->
+
+    <event name="parent" since="3">
+      <description summary="parent change">
+        This event is emitted whenever the parent of the toplevel changes.
+
+        No event is emitted when the parent handle is destroyed by the client.
+      </description>
+      <arg name="parent" type="object" interface="zwlr_foreign_toplevel_handle_v1" allow-null="true"/>
+    </event>
+
+    <!-- Version 4 additions -->
+
+    <event name="desktop_app_id">
```

Comment:
This is kinda the same as the existing `app_id` event, right? The difference being that this will correspond to a *usable* `app_id` more often, because we do some matching in Mir?

Do you know how often applications set a bogus `app_id` on their `xdg_toplevel`s? That is: would it be sensible to pass on the `app_id` of applications that set it, and synthesise an `app_id` for the others, and then use the existing `app_id` event here?

Reply:
They seem to set the `app_id` pretty regularly and when they set it, it's _pretty_ hard to get back to the `.desktop` file from there. For example, opening up `gnome-terminal` reports as `gnome-terminal-server` which in _theory_  I can arrive at `org.gnome.Terminal.desktop` for... but it would be weird given the contents of `org.gnome.Terminal.desktop`

Reply:
And the catch with all of this is that the client would never be able to do the complicated work to resolve that `app_id` into anything usable anyway (especially not without access to a PID for snaps, flatpaks, etc.)